jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/Server.java

```diff
@@ -148,10 +155,21 @@ public String getAlias() {
 		return this.alias;
 	}
 
+	/**
+	 * Returns the pid of the local process jmxtrans will attach to
+	 *
+	 * @return pid of the java process
+	 */
+	public String getPid() {
+		if(this.pid == null) {
+			return null;
+		}
+		return this.pid;
+	}
+
 	public String getHost() {
 		if (host == null && url == null) {
-			// TODO: shouldn't we just return a null in this case ?
-			throw new IllegalStateException("host is null and url is null. Cannot construct host dynamically.");
+			return null;
```

Comment:
I almost did that. However, the port would stay null. This is also why I was talking about getIdentity. That would return host:port or pid@host depending on the use-case